model/task_lifecycle.go

```diff
@@ -2468,10 +2477,19 @@ func checkResetSingleHostTaskGroup(ctx context.Context, t *task.Task, caller str
 		return errors.Errorf("no tasks in task group '%s' for task '%s'", t.TaskGroup, t.Id)
 	}
 	shouldReset := false
+	isAutoRestart := false
 	for _, tgTask := range tasks {
 		if tgTask.ResetWhenFinished {
 			shouldReset = true
 		}
+		if tgTask.IsAutomaticRestart {
+			// Do not restart the task group if it contains a task marked for auto restart
+			// that has since been aborted
+			if tgTask.Aborted {
```

Comment:
Instead of considering aborted in each of these places, can we just unset IsAutomaticRestart when aborting? Or is that more complicated? (Sorry if this has already been asked)

Reply:
Yeah, this logic was put in place because there is a valid case where we [abort a version/build and mark all of its tasks to reset when finished](https://gist.github.com/hadjri/0b567f31c51f78548838c6d691504751#file-db-go-L2853-L2867), but we also need a way to ensure that if we abort a task marked for auto restart before it gets a chance to finish, the scheduled auto restart is ignored.

In lieu of this logic I changed the task abort DB updates to also ensure `ResetWhenFinished` and `IsAutomaticRestart` are marked false, as we should expect those requests to stop the task in its tracks even if it has been marked to reset when finished by an automatic restart. This should achieve the same effect.